### ISSUE TITLE
can not buy from itemshop with untradable item

### ISSUE THREAD

**Issue by sader1992** [rathena/rathena#3481]
* **rAthena Hash**: d17ddf41fe3f1a4eb0fcb75d82888ea89dc697ef

* **Client Date**:  20151104 && 20170614 && 20180502

* **Server Mode**: both

* **Description of Issue**: 
  * Result: can not buy from an `itemshop` if the item have ` 8   - item can't be sold to npcs` in the item_trade db
  * Expected Result: as it's not a normal sell item it should be allowed or at lest the server owner should have the option to allow it
  * How to Reproduce: 
create this shop
`prontera,155,180,0 itemshop Equips 444,6242,501:50`
and try to buy this item `501` he should ask you for 50 `6242`
but instead he refuse to accept it
Because the item trade is
```
// 1   - item can't be dropped
// 8   - item can't be sold to npcs
6242,9,100	// Midgard_Coin
```
* **Modifications that may affect results**: none

https://github.com/rathena/rathena/blob/d17ddf41fe3f1a4eb0fcb75d82888ea89dc697ef/src/map/pc.cpp#L644

we could follow the same as this https://github.com/rathena/rathena/pull/2517/commits/707d1eba4f83893a0a01bdafc51f5f8737d68e07 if we wont allow it all the time
https://github.com/rathena/rathena/blob/d17ddf41fe3f1a4eb0fcb75d82888ea89dc697ef/src/map/pc.cpp#L635

**Comment by cydh** [comment]
actually I don't understand what the issue and what do you want.
because if you want item sold-restricted can be be sold to itemshop if the config is set, is already like that :thinking:

**Comment by sader1992** [comment]
the option in itemtrade should be player sell the item to the npc
but when there is npc sell for the item like itemshop , the server owner would be done that him self and expect it to work
so i think it should at lest a default option to be able to do that

for example >>battleground badges , normally server owners instead of relying on the original npc to sell the items to the player they use item shop instead of delitem and getitem

long story short is selling the item to npc for Xzeny >> the item trade should work on this
but the npc request the item as the (as payment instead of the normal zeny shop) for another item >> the item trade should not work on this

the idea is not selling the item to the npc , the idea is the npc can accept the item as payment it self

maybe the PR is a little miss-leading sorry

**Comment by cydh** [comment]
this config allows the sell restricted item (`0x8` in item_trade) can be sold as currency in `itemshop` https://github.com/rathena/rathena/blob/master/conf/battle/items.conf#L117

**Comment by sader1992** [comment]
sorry i have retest it and turn out it work like it is with changing to `allow_bound_sell: 0x2`
